src/instructlab/data/generate.py

```diff
@@ -210,8 +210,11 @@ def generate(
         from instructlab.model.backends import backends
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
+        # TODO (cdoern): we really should not edit the cfg object
         ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(ctx.obj.config.generate.teacher)
+        backend_instance = backends.select_backend(
+            cfg=ctx.obj.config.serve, model_path=model
```

Comment:
so why do we now ignore `generate.teacher` section? It seems like it no longer used anywhere.

```
$ git grep '\.teacher'
tests/test_config.py:        assert cfg.generate.teacher.model_path == default_model
tests/test_config.py:        assert cfg.generate.teacher.llama_cpp is not None
tests/test_config.py:        assert cfg.generate.teacher.llama_cpp.gpu_layers == -1
tests/test_config.py:        assert cfg.generate.teacher.llama_cpp.max_ctx_size == 4096
tests/test_config.py:        assert cfg.generate.teacher.llama_cpp.llm_family == ""
tests/test_config.py:        assert cfg.generate.teacher.vllm is not None
tests/test_config.py:        assert cfg.generate.teacher.vllm.vllm_args == []
tests/test_config.py:        assert cfg.generate.teacher.host_port == "127.0.0.1:8000"
tests/test_config.py:        assert cfg.generate.teacher.backend is None
```